File: Count and Say.java
class Solution {
    public static String countAndSay(int n) {
        if (n == 1) return "1";

        String s = countAndSay(n - 1);
        StringBuilder sb = new StringBuilder();
        int count = 0;

        for (int i = 0; i < s.length(); i++) {
            count++;

            if (i + 1 >= s.length() || s.charAt(i) != s.charAt(i + 1)) {
                sb.append(count);
                sb.append(s.charAt(i));
                count = 0;
            }
        }

        return sb.toString();
    }
}

File: Count_and_Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1){
            return "1";
        }
        string s = countAndSay(n-1);
        char cur = s[0];
        int c=0;
        string ans= "";
        for(int i=0;i<s.length();i++){
            if(cur == s[i]){
                c++;
            }
            else{
                ans += c+'0';
                ans += cur;
                cur= s[i];
                c=1;
            }
        }
        ans+= c+'0';
        ans += cur;
        return ans;
    }
};
